File: app/src/main/java/cn/devifish/readme/view/base/BaseFragment.java
package cn.devifish.readme.view.base;

import android.os.Bundle;
import android.support.annotation.LayoutRes;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import butterknife.ButterKnife;

/**
 * Created by zhang on 2017/4/5.
 *
 * Fragment基类
 */

public abstract class BaseFragment extends Fragment {

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View rootView = inflater.inflate(bindLayout(), container, false);
        ButterKnife.bind(this, rootView);

        initVar();
        initView();
        return rootView;
    }

    @LayoutRes
    protected abstract int bindLayout();
    protected abstract void initVar();
    protected abstract void initView();

}

File: app/src/main/java/cn/devifish/readme/view/base/BaseFragment.kt
package cn.devifish.readme.view.base

import android.os.Bundle
import android.support.annotation.LayoutRes
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

/**
 * Created by zhang on 2017/6/3.
 * Fragment基类
 */
abstract class BaseFragment : Fragment() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
    }

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        if (inflater != null && bindLayout() != 0) {
            return inflater.inflate(bindLayout(), container, false)
        }
        return super.onCreateView(inflater, container, savedInstanceState)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        initVar()
        initView(view)
    }

    @LayoutRes
    protected abstract fun bindLayout(): Int

    protected abstract fun initVar()
    protected abstract fun initView(view: View)

}